src/main/java/toothless/parser/Parser.java

```diff
@@ -0,0 +1,183 @@
+package toothless.parser;
+
+import java.util.ArrayList;
+
+import toothless.exception.ToothlessException;
+import toothless.task.Task;
+import toothless.task.TaskList;
+import toothless.ui.Ui;
+
+/**
+ * A class that parses, resolves and validates user input.
+ */
+public class Parser {
+    /**
+     * The main method that parses user input.
+     * @param userInput User input string to be parsed.
```

Comment:
Should there be a newline separating the description of the javadoc with the parameter section? This would better adhere to the coding standards [here](https://se-education.org/guides/conventions/java/intermediate.html#comments)

I see this problem in your other javadocs as well